### PR TITLE
stacktrace: Remove addrmap item on release

### PR DESCRIPTION
**Pull request by Bojun-Seo**
Remove addrmap item on release
Remove unnecessary(duplicated) find operation on erasing stackmap item
Add unlikely macro for performance

### CODE REVIEW

File: stacktrace.cc [hunk 1]
```diff
@@ -69,24 +69,23 @@ void release_backtrace(void* addr)
...
 	stack_info_t& stack_info = stackit->second;
 	stack_info.total_size -= object_info.size;
 	stack_info.count--;
 	if (stack_info.count == 0) {
-		const auto& it = stackmap.find(stack_trace);
-		if (it != stackmap.end())
-			stackmap.erase(it);
+		stackmap.erase(stackit);
 	}
+	addrmap.erase(addrit);
```
Comment (honggyukim) [inline stacktrace.cc:88]: I would like to suggest this part as follows.
```c
	if (stack_info.count == 0) {
		// The stackmap for the given stacktrace is no longer needed.
		stackmap.erase(stackit);
	}

	// The given address is released so remove it from addrmap.
	addrmap.erase(addrit);
```

**Review by honggyukim — COMMENTED** [review]
Thanks for the patch. It looks good but the current commit includes multiple unrelated changes so please split the commit into 3 separate pieces as follows.
1. Add `unlikely` for branch optimization
2. Remove duplicated `stackmap.find` for erase
3. Erase address from `addrmap` when the address is released

**Comment by honggyukim** [comment]
And please correct your author name in git config. Your name is shown as just "Bojun" as follows.
```
$ git shortlog -ns
   118  Honggyu Kim
     2  Bojun
     1  Soim
```
You can easily fix the name with this.
```
$ git config --global user.name "Bojun Seo"
```
Please do that before updating your commit.

**Comment by Bojun-Seo** [comment]
Thanks for your opinions.
I'll split commits and change author name as well.

File: stacktrace.cc (hunk 1 above)
Comment (Bojun-Seo) [reply stacktrace.cc]: I changed the patch as you recommended

**Review by honggyukim — APPROVED** [review]
Thanks for the update! LGTM.

**Pull request merged**